content/en/docs/tasks/debug-application-cluster/local-debugging.md

```diff
@@ -2,60 +2,67 @@
-
-Use the `--swap-deployment` option to swap an existing deployment with the Telepresence proxy. Swapping allows you to run a service locally and connect to the remote Kubernetes cluster. The services in the remote cluster can now access the locally running instance.
-
-To run telepresence with `--swap-deployment`, enter:
-
-`telepresence --swap-deployment $DEPLOYMENT_NAME`
-
-where $DEPLOYMENT_NAME is the name of your existing deployment.
-
-Running this command spawns a shell. In the shell, start your service. You can then make edits to the source code locally, save, and see the changes take effect immediately. You can also run your service in a debugger, or any other local development tool.
-
-
-
+ 
+Use the `telepresence intercept $SERVICE_NAME --port $LOCAL_PORT:REMOTE_PORT` command to create an "intercept" for rerouting remote service traffic.
+ 
+Where:
+ 
+-   `$SERVICE_NAME`  is the name of your local service
+-   `$LOCAL_PORT` is the port that your service is running on your local workstation
+-   And `$REMOTE_PORT` is the port your service listens to in the cluster
+ 
+Running this command will tell Telepresence to send remote traffic to your local service instead of the service in the remote Kubernetes cluster.  With this, you can then make edits to your service source code locally, save, and see the corresponding changes when accessing your remote application take effect immediately. You can also run your local service using a debugger or any other local development tool.
+ 
+<!-- how telepresence works -->
+ 
+## How does Telepresence work?
+ 
+Telepresence installs a traffic-agent sidecar next to your existing application's container running in the remote cluster. It then captures all traffic requests going into the Pod, and instead of forwarding this to the application in the remote cluster, it routes all traffic (when you create a [global intercept](https://www.getambassador.io/docs/telepresence/latest/concepts/intercepts/#global-intercept)) or a subset of the traffic (when you create a [personal intercept](https://www.getambassador.io/docs/telepresence/latest/concepts/intercepts/#personal-intercept)) to your local development environment.
+ 
 ## {{% heading "whatsnext" %}}
-
-
-If you're interested in a hands-on tutorial, check out [this tutorial](https://cloud.google.com/community/tutorials/developing-services-with-k8s) that walks through locally developing the Guestbook application on Google Kubernetes Engine.
-
-Telepresence has [numerous proxying options](https://www.telepresence.io/reference/methods), depending on your situation.
-
+ 
+If you're interested in a hands-on tutorial, check out [this tutorial](https://blog.jetbrains.com/idea/2021/05/easily-debug-java-microservices-running-on-kubernetes-with-intellij-idea/) that walks through locally debugging Java microservices with IntelliJ IDEA and Telepresence connected to a remote Kubernetes cluster.
+ 
```

Comment:
I'm not sure if we need this in the Kubernetes documentation since we're already linking to the Telepresence website if folks want to learn more. @sftim @reylejano WDYT?

Reply:
Thanks for the feedback @RinkiyaKeDad. If you check the original [document](https://kubernetes.io/docs/tasks/debug-application-cluster/local-debugging/#what-s-next), you will see it actually includes a tutorial about using Telepresence with Google Kubernetes Engine's Guestbook applications. I had initially replaced it with another article because it was outdated. 

Now, that I have sent a PR to the Google Cloud team, I have reverted this document to the original article since it will be up to date when my PR is merged by the Google Cloud team. Let me know what you think.

Reply:
Hello @sftim and @reylejano. Can you please share your thoughts on this? So, I can make the necessary changes (if any) and get the pull request merged. Thank you. 🙏